sdk/appconfiguration/app-configuration/test/public/index.spec.ts

```diff
@@ -851,6 +851,107 @@ describe("AppConfigurationClient", () => {
       }
     });
 
+    it.only("list with multiple pages - bypage and etags", async function () {
+      // This occasionally hits 429 error (throttling) since we are making 100s of requests in the test to create, get and delete keys.
+      // To avoid hitting the service with too many requests, skipping the test in live.
+      // More details at https://github.com/Azure/azure-sdk-for-js/issues/16743
+      //
+      // Remove the following line if you want to hit the live service.
+      // eslint-disable-next-line @typescript-eslint/no-invalid-this
+      // if (isLiveMode()) this.skip();
+
+      const key = recorder.variable(
+        "listMultiplePagesOfResults",
+        `listMultiplePagesOfResults${Math.floor(Math.random() * 1000)}`,
+      );
+
+      // this number is arbitrarily chosen to match the size of a page + 1
+      const expectedNumberOfLabels = 200;
+
+      let addSettingPromises = [];
+
+      for (let i = 0; i < expectedNumberOfLabels; i++) {
+        addSettingPromises.push(
+          client.addConfigurationSetting({
+            key,
+            value: `the value for ${i}`,
+            label: i.toString(),
+          }),
+        );
+
+        if (i !== 0 && i % 2 === 0) {
+          await Promise.all(addSettingPromises);
+          addSettingPromises = [];
+        }
+      }
+
+      await Promise.all(addSettingPromises);
+
+      // Passing marker as an argument
+      let iterator = client.listConfigurationSettings({ keyFilter: key, }).byPage();
+      const etags: string[] = [];
+      for await (const page of iterator) {
+        console.log(`  page: ${JSON.stringify(page)}`);
+        etags.push(page.etag ?? "");
+      }
+
+      // Add 50 more settings
+      addSettingPromises = [];
+
+      for (let i = expectedNumberOfLabels; i < expectedNumberOfLabels * 5 / 4; i++) {
```

Comment:
Is there a particular reason why we have to do math instead of just `+ 50` 😄

Reply:
In case we change the number of labels, we can just edit the centralized `expectedNumberOfLabels` and everything would work as expected.

But in this case though, it doesn't matter since adding any number of settings would cascade the pages and shift the etags.